backends/gstreamer/src/audio_sink.rs

```diff
@@ -95,38 +109,42 @@ impl AudioSink for GStreamerAudioSink {
             .unwrap();
 
         let appsrc = self.appsrc.as_ref().clone().upcast();
-        let resample = gst::ElementFactory::make("audioresample", None).ok_or(())?;
-        let convert = gst::ElementFactory::make("audioconvert", None).ok_or(())?;
-        let sink = gst::ElementFactory::make("autoaudiosink", None).ok_or(())?;
+        let resample = gst::ElementFactory::make("audioresample", None)
+            .ok_or(BackendError::ElementCreationFailed("audioresample"))?;
+        let convert = gst::ElementFactory::make("audioconvert", None)
+            .ok_or(BackendError::ElementCreationFailed("audioconvert"))?;
+        let sink = gst::ElementFactory::make("autoaudiosink", None)
+            .ok_or(BackendError::ElementCreationFailed("autoaudiosink"))?;
         self.pipeline
             .add_many(&[&appsrc, &resample, &convert, &sink])
-            .map_err(|_| ())?;
-        gst::Element::link_many(&[&appsrc, &resample, &convert, &sink]).map_err(|_| ())?;
+            .map_err(|_| BackendError::PipelineFailed)?;
+        gst::Element::link_many(&[&appsrc, &resample, &convert, &sink])
+            .map_err(|_| BackendError::PipelineFailed)?;
```

Comment:
You maybe want to take the `&'static str` out of the error and store it in `PipelineFailed`